nomad/core_sched.go

```diff
@@ -286,7 +288,13 @@ func (c *CoreScheduler) gcEval(eval *structs.Evaluation, thresholdIndex uint64,
 	gcEval := true
 	var gcAllocIDs []string
 	for _, alloc := range allocs {
-		if !alloc.TerminalStatus() || alloc.ModifyIndex > thresholdIndex {
+		var reschedulePolicy *structs.ReschedulePolicy
+		tg := job.LookupTaskGroup(alloc.TaskGroup)
+
+		if tg != nil {
+			reschedulePolicy = tg.ReschedulePolicy
+		}
+		if !alloc.GCEligible(reschedulePolicy, time.Now(), thresholdIndex) {
```

Comment:
Is there a reason why the current time needs to be injected as a parameter?

Reply:
for easier mocking in tests, and in the future we might want to replace current time with the eval's last modified time or something like that.